modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java

```diff
@@ -34,22 +39,54 @@
 public class EndpointHelper {
 
 	public Map<String, Object> getResponseEntityMap(
-			long companyId, APIApplication.Schema schema,
-			String pathParameterValue)
+			long companyId, APIApplication.Schema schema, String pathParameter,
+			String pathParameterValue, String scopeKey)
 		throws Exception {
 
 		Set<String> relationshipsNames = new HashSet<>();
 
+		ObjectEntry objectEntry = null;
+
 		for (APIApplication.Property property : schema.getProperties()) {
 			relationshipsNames.addAll(property.getObjectRelationshipNames());
 		}
 
-		return _getResponseEntityMap(
-			_objectEntryHelper.getObjectEntry(
+		if (Objects.equals(
+				pathParameter, HeadlessBuilderConstants.PATH_PARAMETER_ID)) {
+
+			objectEntry = _objectEntryHelper.getObjectEntry(
 				companyId, ListUtil.fromCollection(relationshipsNames),
 				GetterUtil.getLong(pathParameterValue),
-				schema.getMainObjectDefinitionExternalReferenceCode()),
-			schema);
+				schema.getMainObjectDefinitionExternalReferenceCode());
+		}
+		else if (Objects.equals(
+					pathParameter,
+					HeadlessBuilderConstants.PATH_PARAMETER_ERC)) {
+
+			objectEntry =
+				_objectEntryHelper.getObjectEntryByExternalReferenceCode(
+					companyId, ListUtil.fromCollection(relationshipsNames),
+					schema.getMainObjectDefinitionExternalReferenceCode(),
+					pathParameterValue, scopeKey);
+		}
+		else {
+			List<ObjectEntry> objectEntries =
+				_objectEntryHelper.getObjectEntries(
+					companyId,
+					StringBundler.concat(
+						pathParameter, " eq '", pathParameterValue, "'"),
+					ListUtil.fromCollection(relationshipsNames),
+					schema.getMainObjectDefinitionExternalReferenceCode(),
+					scopeKey);
+
+			if (objectEntries.isEmpty()) {
+				throw new NotFoundException();
```

Comment:
I am not sure if we are testing this result. Anyway, I am worried about throwing a different exception than in the other two cases (the one by id and the one by ERC). Could you verify?

Reply:
In the case for id and ERC, it throws a 404 code with response body from the Objects side, but as we are making a request using a filter expression from the collection endpoints, if the item list is empty, we have to manage a not found exception

Reply:
As well as, it's true that we should test this scenario